OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/immutability/L0FieldImmutabilityAnalysis.scala

```diff
@@ -0,0 +1,429 @@
+import org.opalj.br.ClassTypeSignature
+import org.opalj.br.Field
+import org.opalj.br.ObjectType
+import org.opalj.br.ProperTypeArgument
+import org.opalj.br.RuntimeInvisibleAnnotationTable
+import org.opalj.br.SimpleClassTypeSignature
+import org.opalj.br.SourceFile
+import org.opalj.br.TypeVariableSignature
+import org.opalj.br.analyses.ProjectInformationKeys
+import org.opalj.br.analyses.SomeProject
+import org.opalj.br.fpcf.BasicFPCFEagerAnalysisScheduler
+import org.opalj.br.fpcf.BasicFPCFLazyAnalysisScheduler
+import org.opalj.br.fpcf.FPCFAnalysis
+import org.opalj.br.fpcf.FPCFAnalysisScheduler
+import org.opalj.br.ReferenceType
+import org.opalj.br.fpcf.properties.immutability.Assignable
+import org.opalj.br.fpcf.properties.immutability.ClassImmutability
+import org.opalj.br.fpcf.properties.immutability.DependentlyImmutableClass
+import org.opalj.br.fpcf.properties.immutability.DependentlyImmutableField
+import org.opalj.br.fpcf.properties.immutability.DependentlyImmutableType
+import org.opalj.br.fpcf.properties.immutability.EffectivelyNonAssignable
+import org.opalj.br.fpcf.properties.immutability.FieldAssignability
+import org.opalj.br.fpcf.properties.immutability.FieldImmutability
+import org.opalj.br.fpcf.properties.immutability.LazilyInitialized
+import org.opalj.br.fpcf.properties.immutability.MutableClass
+import org.opalj.br.fpcf.properties.immutability.MutableField
+import org.opalj.br.fpcf.properties.immutability.MutableType
+import org.opalj.br.fpcf.properties.immutability.NonAssignable
+import org.opalj.br.fpcf.properties.immutability.NonTransitivelyImmutableClass
+import org.opalj.br.fpcf.properties.immutability.NonTransitivelyImmutableField
+import org.opalj.br.fpcf.properties.immutability.NonTransitivelyImmutableType
+import org.opalj.br.fpcf.properties.immutability.TransitivelyImmutableClass
+import org.opalj.br.fpcf.properties.immutability.TransitivelyImmutableField
+import org.opalj.br.fpcf.properties.immutability.TransitivelyImmutableType
+import org.opalj.br.fpcf.properties.immutability.TypeImmutability
+import org.opalj.br.fpcf.properties.immutability.UnsafelyLazilyInitialized
+import org.opalj.tac.cg.TypeIteratorKey
+import org.opalj.tac.fpcf.analyses.cg.BaseAnalysisState
+import org.opalj.tac.fpcf.analyses.cg.TypeIterator
+import org.opalj.tac.fpcf.analyses.cg.TypeIteratorState
+import org.opalj.br.analyses.cg.TypeExtensibilityKey
+import org.opalj.br.FieldType
+import org.opalj.br.ConstantFloat
+import org.opalj.br.ConstantInteger
+import org.opalj.br.ConstantString
+import org.opalj.br.ConstantLong
+import org.opalj.br.ArrayTypeSignature
+import org.opalj.br.Wildcard
+import org.opalj.br.ConstantDouble
+import org.opalj.br.Synthetic
+import org.opalj.br.RuntimeVisibleAnnotationTable
+import org.opalj.br.Deprecated
+
+/**
+ * Analysis that determines the immutability of org.opalj.br.Field
+ * @author Tobias Roth
+ */
+class L0FieldImmutabilityAnalysis private[analyses] (val project: SomeProject)
```

Comment:
Why is this analysis called `L0`? Is there any reason to ever expect there to be a more precise variant of this analysis? Since it's just an aggregating analysis, I would have assumed this to be the definitive implementation and, thus, not have a level specifier.